game/src/ecs/components/skill/Skill.java

```diff
@@ -1,35 +1,35 @@
 package ecs.components.skill;
 
+import ecs.entities.Entity;
 import graphic.Animation;
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 
 public class Skill {
 
-    private Method method;
+    private Entity entity;
     private boolean active;
     private Animation animation;
+    private ISkillFunction skillFunction;
 
     /**
-     * @param method Method with the logic of this skill
+     * @param entity associated entity
      * @param animation Animation of this skill
+     * @param skillFunction Function of this skill
      */
-    public Skill(Method method, Animation animation) {
-        this.method = method;
+    public Skill(Entity entity, Animation animation, ISkillFunction skillFunction) {
```

Comment:
Bin grade am Überlegen, ich sehe zwei Optionen:

Option A: Ein Skill gehört genau zu einer Entity. Dann braucht es hier den Parameter `entity` aber in `execute` nicht mehr.

Option B: Ein Skill kann von mehreren Entitäten benutzt werden. Dann braucht es hier Parameter und Attribut nicht mehr. In `execute` wird dann die Entität übergeben, die den Skill ausführt.


B mach für mich einen schöneren Eindruck. Dann kann jedes Monster den gleichen Feuerball spucken. 
@Lena241 Meinung?

Reply:
B macht auf mich auch den besseren Eindruck.